mongo/problem/problem.py

```diff
@@ -486,6 +487,39 @@ def get_test_case(self) -> BinaryIO:
         # fallback to legacy GridFS storage
         return self.test_case.case_zip
 
+    def migrate_gridfs_to_minio(self):
+        '''
+        migrate test case from gridfs to minio
+        '''
+        if self.test_case.case_zip.grid_id is None:
+            return
+
+        if self.test_case.case_zip_minio_path is not None:
+            self.check_test_case_consistency()
+            return
+
+        # migrate to minio
+        self._save_test_case_zip(self.test_case.case_zip)
+        self.test_case.case_zip.delete()
```

Comment:
I think we can make `check_test_case_consistency` return bool to indicate if consistent or not, and in `migrate_gridfs_to_minio`, we can use this return value to determine if to call delete()

```
    def migrate_gridfs_to_minio(self):
        '''
        migrate test case from gridfs to minio
        '''
        if self.test_case.case_zip.grid_id is None:
            return

        if self.test_case.case_zip_minio_path is not None:
            if self.check_test_case_consistency():
                self.test_case.case_zip.delete()
            return

        # migrate to minio
        # I don't know why we save here
        self._save_test_case_zip(self.test_case.case_zip)
        if self.check_test_case_consistency():
            self.test_case.case_zip.delete()
```

Reply:
make sense. I have updated the method.